OPHD/StructureCatalogue.cpp

```diff
@@ -29,11 +30,32 @@ Structure* StructureCatalogue::get(StructureTypeID type)
 	// This seems like a naive approach... I usually see these implemented as the base
 	// object type has a static function that is used as an interface to instantiate
 	// derived types.
+	// TODO: (#843) Pulling component-based structure component in here for now.
+	//       This class has friend access to Structure and can therefore call the various
+	//       initialization functions. This will be replaced by a generic factory function
+	//       that deserializes descriptions from the structure catalog and save files.
 	switch (type)
 	{
 		case StructureTypeID::SID_AGRIDOME:
-			structure = new Agridome();
-			break;
+		{
+			structure = new Structure(
+				constants::AGRIDOME,
+				"structures/agridome.sprite",
+				Structure::StructureClass::FoodProduction,
+				StructureTypeID::SID_AGRIDOME);
+			structure->maxAge(600);
+			structure->turnsToBuild(3);
+			structure->requiresCHAP(true);
+			structure->resourcesIn({ 1, 0, 0, 0 });
+			structure->energyRequired(2);
+
+			const int AGRIDOME_CAPACITY = 1000;
+			const int AGRIDOME_BASE_PRODUCUCTION = 10;
+			FoodProduction* foodProduction = new FoodProduction(structure);
+			foodProduction->initialize(AGRIDOME_BASE_PRODUCUCTION, AGRIDOME_CAPACITY);
+			NAS2D::Utility<StructureManager>::get().attachComponent(structure, foodProduction);
```

Comment:
this is quite ugly, but temporary. The initialize call will be replaced by structure type deserialization as discussed in #843 . I also dislike calling `StructureManager::attachComponent` on a `Structure*` that has not yet been added with `StructureManager::addStructure`, but this is the only place I have the actual component type. Again, this won't be an issue once #843 is addressed.

Reply:
I considered putting this code in a temporary `CreateAgridome` function to avoid the temporary clutter in `StructureCatalogue::get` but a bunch of the `Structure` members are protected. It seemed less offensive to put this here than to make the members public or giving friend access to a bunch of structure factory functions.

Reply:
This is what I'd call 'falsework' (from construction lingo) and is acceptable. It makes sense to me to have something temporarily in place until it can be replaced with something better. It's the nature of iterative work I think.